src/Orders/OrderDetailsScreen.js

```diff
@@ -0,0 +1,358 @@
+      filterByFormula: `({email}='${useremail}')`,
+    }).firstPage()
+      .then((records) => {
+        if (`${records[0].fields['apartment number']}` !== '') {
+          setShippingAddress({
+            address: records[0].fields.address,
+            zipcode: records[0].fields.zipcode,
+            apartmentLine: ` Apt ${records[0].fields['apartment number']}`,
+          });
+        } else {
+          setShippingAddress({
+            address: records[0].fields.address,
+            zipcode: records[0].fields.zipcode,
+            apartmentLine: '',
+          });
+        }
+      });
+  };
+
+  // Helper function to wait for [ms] milliseconds
+  function timer(ms) { return new Promise((res) => setTimeout(res, ms)); }
+
+  // Batch function to serialize airtable calls (since api is rate limited at
+  // 10 row updates at a time)
+  const cartBatch = async (cartObj) => { // 3
+    base('CART V3').create(cartObj, (err) => {
+      if (err) {
+        console.error(err);
+      }
+    });
+    await timer(1000);
+  };
+
+  // Add current items to airtable in the Cart table
+  const orderAgain = async () => {
+    const cartBatches = [];
+    for (let j = 0; j < Math.floor(items[1].length / 10) + 1; j += 1) {
+      const cartObj = [];
+      let upper = items[1].length % 10;
+
+      if (j < Math.floor(items[1].length / 10)) {
+        upper = 10;
+      }
+      for (let i = 0; i < upper; i += 1) {
+        const cartRow = {};
+        cartRow.fields = {
+          Produce: items[1][i].produce_id,
+          shopper: [CONST_USER_ID],
+          quantity: items[1][i].Quantity,
+        };
+        cartObj.push(cartRow);
+      }
+      cartBatches.push(cartObj);
+    }
+
+    for (let i = 0; i < cartBatches.length; i += 1) {
+      await cartBatch(cartBatches[i]);
+    }
+  };
+
+  useEffect(() => {
+    // TODO: replace hardcoded email with logged in user info
+    setOrderDetails(CONST_USER);
+    setDeliveryFee(10);
+    setFreeFee(20);
+    setCount(items[1].length);
+    const regItemList = [];
+    let totalPrice = 0;
+    const quantities = [];
+
+    items[1].forEach((item) => {
+      base('Produce').find(item.produce_id, ((err, prodRecord) => {
+        if (err) { console.error(err); return; }
+        const prodItem = prodRecord;
+        regItemList.push(prodItem.fields);
+        quantities.push(String(item.Quantity));
+        totalPrice += parseFloat(item.Quantity) * parseFloat(prodItem.fields.Price);
+        setTotal(totalPrice);
+        if (regItemList.length === items[1].length) {
+          const products = regItemList.map((prodItemIter, ind) => (
+            <OrderItem
+              name={prodItemIter.Name}
+              price={prodItemIter.Price}
+              type={prodItemIter.Unit}
+              quantity={quantities[ind]}
+              key={ind}
+              image={prodItemIter.Image[0].url}
+            />
+          ));
+          setProductList(products);
+        }
+      }));
+    });
+  }, []);
```

Comment:
what do these lines do? ESlint throws one "error" about line 219 using ind (an array index) as the key when rendering react components. But reading about it online, it seems like its only really an issue if we delete any of the components. Assuming this code generates the OrderCards, I think it's fine...

Reply:
Yeah this map just creates an js array of every produce item for the order details screen. Eslint throws a different error if I don't specify a unique key at all, which is why I included line 219. I think since it works, it should be fine?